### ISSUE TITLE
Subsequent savestate loading needs F11+L to be pressed twice after first load

### ISSUE THREAD

**Issue by rebb** [joncampbell123/dosbox-x#3063]
### Code of Conduct & Contributing Guidelines

- [X] I agree to follow the code of conduct and the contributing guidelines.

### Have you checked that no other similar bug report(s) already exists?

- [X] I have searched and didn't find any similar issues.

### What operating system(s) this bug have occurred on?

Win7

### What version(s) of DOSBox-X have this bug?

0.83.19

### Describe the bug

Subsequent savestate loads after the first load of a savestate require F11+L to be pressed twice before anything happens.

Not a huge issue, just annoying in situations when one is (ab)using the savestate system very quickly ;)

### Expected behavior

Pressing F11+L should always trigger savestate loading

### Steps to reproduce the behaviour

1. Make a savestate ( F11+S )
2. Do something in the game so you will notice the savestate change
3. Load savestate ( F11+L ) - Note that the first load after the savestate always loads immediately ( as it should )
4. Do something
5. Press F11+L again - Nothing happens
6. Press F11+L again - It loads
7. Press F11+L again - Nothing happens
8. Press F11+L again - It loads
7. This behaviour repeats until a new savestate is made

**Comment by Wengier** [comment]
Looks like an issue specific to SDL1 builds and `aspect=true`. Please try the SDL2 builds which will hopefully work.

**Comment by rebb** [comment]
You're right it seems to work in the SLD2 build.
I always thought the SDL1 build was considered the preferred one though, since it allows deeper customization of the code.

Is that still the case or are both builds pretty much the same quality and feature wise ?

**Comment by Wengier** [comment]
@rebb The most notable differences between the Windows SDL1 and SDL2 builds are the menu style, which is the result of deeper customization of the code. But other than that I think they are pretty similar feature-wise indeed. Meanwhile, the SDL2 build has improved input handling sometimes, such as this one.

**Comment by rebb** [comment]
Interesting, thanks :)
Not sure if this issue should remain open or not, i annotated it with SDL1-only in any case.

**Comment by rebb** [comment]
Unrelated, but for some reason i'm getting much more frequent savestate corruption when using the SDL2 build.
Maybe i'm just unlucky though. Weird.

**Comment by Wengier** [comment]
@rebb How about the MinGW SDL2 builds then?

**Comment by rebb** [comment]
Not sure where to get that, i only see a MingW-HX-DOS build in for the current release.
Or is it that one ?

**Comment by Wengier** [comment]
@rebb No, MinGW builds are different and are available in the Windows installer.

**Comment by rebb** [comment]
Oh i see, i always use the portable packages from Github. I'll check it out then.

**Comment by rebb** [comment]
So far i've not seen any signs of memory corruption using the MinGW SDL2 build, and i've done a good bunch of savestating - so that's a good sign :)

Does the MinGW compiler do anything differently related to savestates ?

**Comment by Wengier** [comment]
@rebb I don't think there is dedicated code to do special handling for save states in MinGW builds, but according to issue #1986 MinGW builds do indeed support save-states better for some reason.

**Comment by rebb** [comment]
Just as an update, there seems to be a bit of crashing going on with savestate loading in the MingW SDL2 build. Not sure if it's the same core issue that causes corruption in the other build. Maybe in MingW it crashes instead of corrupting things ? Just a wild guess though.
